tools/PI/DevHome.PI/ViewModels/ClipboardMonitorControlViewModel.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.ComponentModel;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using DevHome.PI.Models;
+
+namespace DevHome.PI.ViewModels;
+
+public partial class ClipboardMonitorControlViewModel : ObservableObject
+{
+    private readonly Microsoft.UI.Dispatching.DispatcherQueue _dispatcher;
+
+    [ObservableProperty]
+    private string _clipboardContentsHex = string.Empty;
+
+    [ObservableProperty]
+    private string _clipboardContentsDec = string.Empty;
+
+    [ObservableProperty]
+    private string _clipboardContentsCode = string.Empty;
+
+    [ObservableProperty]
+    private string? _clipboardContentsHelp;
+
+    private bool _listenForClipboardChanges = true;
+
+    public ClipboardMonitorControlViewModel()
+    {
+        _dispatcher = Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread();
```

Comment:
In the rest of Dev Home it's not good to use this, since we may not be on the UI thread, and therefore this may return null. Is different in PI?

Reply:
Yeah, in PI we always create the view model in the view's constructor. When does DevHome create its view models?

Reply:
Sometimes in view's constructor, but often in services registered with the host.

Reply:
Ahh... so your services push data to a singleton view models vs having the view models pull data from the services?

We did go down that path a bit, but since our views can come and go, it became a pain to "retarget" the view model to new views when they went away. So now our services/models maintain state, and view models can come and go and query that/register callbacks/etc when needed.